plugins/inputs/powerdns/powerdns.go

```diff
@@ -49,59 +49,105 @@ func (p *Powerdns) Gather(acc telegraf.Accumulator) error {
 }
 
 func (p *Powerdns) gatherServer(address string, acc telegraf.Accumulator) error {
-	conn, err := net.DialTimeout("unix", address, defaultTimeout)
+	stats_cmd := "show * \n"
+	// fetch general stats
+	stats, err := p.executeCommand(stats_cmd, address)
 	if err != nil {
 		return err
 	}
+	// Process general stats data
+	fields := parseStatsResponse(stats)
+	// Add server socket as a tag
+	tags := map[string]string{"server": address}
+	acc.AddFields("powerdns", fields, tags)
 
-	defer conn.Close()
+	// fetch respsizes stats for calculating histogram metrics
+	respsizes_cmd := "respsizes \n"
+	respsizes, err := p.executeCommand(respsizes_cmd, address)
+	if err != nil {
+		return err
+	}
+	// Process latency stats data
+	latencyBucket := parseRespsizesResponse(respsizes)
+	for latency, count := range latencyBucket {
+		tags["latency"] = strconv.Itoa(int(latency))
+		fields := map[string]interface{}{
+			"resp_size": count,
+		}
+		acc.AddCounter("powerdns", fields, tags)
+	}
+	return nil
+}
 
+func (p *Powerdns) executeCommand(command, address string) (string, error) {
+	conn, err := net.DialTimeout("unix", address, defaultTimeout)
```

Comment:
It would be ideal if we could only dial once for both commands, is this change possible?